### PR TITLE
Add procenv

### PR DESCRIPTION
**Pull request by loveshack**

### CODE REVIEW

**Review by michaelkuhn — CHANGES_REQUESTED** [review]
There is one flake8 error that has to be fixed (missing newline above the class).

Moreover, procenv currently does not compile with GCC 10, so we should add a conflict or apply the following patch: https://github.com/jamesodhunt/procenv/pull/16

**Comment by michaelkuhn** [comment]
I think you forgot to actually add the patch. You can also just do something like this:
```
patch('https://github.com/jamesodhunt/procenv/pull/16.patch', sha256='7a2b34c754a80ba5fde53626290f9955ec341cd00dbeb45212e6e20d617c7029')
```

**Comment by loveshack** [comment]
I realized this was outstanding. I did add the patch.  Thanks for the hint about using a url, which I didn't know.

File: var/spack/repos/builtin/packages/procenv/package.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Procenv(AutotoolsPackage):
+    """A command-line tool that displays as much detail about itself and
+its environment as possible.  It can be used as a test tool, to
+understand the type of environment a process runs in, and for
+comparing system environments."""
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/procenv/package.py]: Can you indent this?

**Review by michaelkuhn — APPROVED** [review]
LGTM (once Adam's change request has been incorporated)

**Pull request merged**